test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1

```diff
@@ -0,0 +1,653 @@
+
+        New-Item -Path "subDir2" -ItemType Directory > $null
+        Set-Location "subDir2"
+        New-Item -Path "file2.txt" -ItemType File > $null
+        (New-Item -Path "filehidden2.asd" -ItemType File).Attributes = "Hidden"
+        (New-Item -Path "filereadonly2.doc" -ItemType File).Attributes = "ReadOnly"
+        (New-Item -Path "subDir21" -ItemType Directory).Attributes = "Hidden"
+        Set-Location "subDir21"
+        New-Item -Path "file21.txt" -ItemType File > $null
+
+        Set-Location $rootDir
+        New-Item -Path "subDir3" -ItemType Directory > $null
+        Set-Location "subDir3"
+        New-Item -Path "file3.asd" -ItemType File > $null
+        (New-Item -Path "filehidden3.txt" -ItemType File).Attributes = "Hidden"
+        (New-Item -Path "filereadonly3.doc" -ItemType File).Attributes = "ReadOnly"
+
+        Set-Location $rootDir
+    }
+
+    AfterAll {
+        #restore the previous location
+        Set-Location -Path $restoreLocation
+    }
+
+    Context 'Validate Get-ChildItem -Path' {
+        It "Get-ChildItem -Path" {
+            $result = Get-ChildItem -Path $rootDir
+            $result.Count | Should -Be 4
+            $result[0] | Should -BeOfType System.IO.DirectoryInfo
+        }
+
+        It "Get-ChildItem -Path -Hidden" {
+            $result = Get-ChildItem -Path $rootDir -Hidden
+            $result.Count | Should -Be 1
+            $result | Should -BeOfType System.IO.FileInfo
+            $result.Name | Should -BeExactly "filehidden1.doc"
+        }
+
+        It "Get-ChildItem -Path -Attribute Hidden" {
+            $result = Get-ChildItem -Path $rootDir -Attributes Hidden
+            $result.Count | Should -Be 1
+            $result | Should -BeOfType System.IO.FileInfo
+            $result.Name | Should -BeExactly "filehidden1.doc"
+        }
+
+        It "Get-ChildItem -Path -Force" {
+            $result = Get-ChildItem -Path $rootDir -Force
+            $result.Count | Should -Be 5
+            $result.Name | Should -Contain "filehidden1.doc"
+        }
+    }
+
+    Context 'Validate Get-ChildItem -Path -Directory/-File' {
+        It "Get-ChildItem -Path -Directory" {
+            $result = Get-ChildItem -Path $rootDir -Directory
+            $result.Count | Should -Be 2
+        }
+
+        It "Get-ChildItem -Path -File" {
+            $result = Get-ChildItem -Path $rootDir -File
+            $result.Count | Should -Be 2
+        }
+
+        It "Get-ChildItem -Path -File -Hidden" {
+            $result = Get-ChildItem -Path $rootDir -File -Hidden
+            $result.Count | Should -Be 1
+            $result | Should -BeOfType System.IO.FileInfo
+            $result.Name | Should -BeExactly "filehidden1.doc"
+        }
+    }
+
+    Context 'Validate Get-ChildItem -Path -Name' {
+        It "Get-ChildItem -Path -Name" {
+            $result = Get-ChildItem -Path $rootDir -Name
+            $result.Count | Should -Be 4
+            $result[0] | Should -BeOfType System.String
```

Comment:
NIT: Although a string is accepted, I think `[System.String]` makes it stand out a bit as a type

Reply:
It was discussed in #11558 with the conclusion to use the Pester pattern https://pester.dev/docs/usage/assertions/#beoftype